### ISSUE TITLE
Transposing two times

### ISSUE THREAD

**Issue by juliagarriga** [marcromani/nica-nmf#1]
Wouldn't it be better to be, in function `nica_nmf` at `cocktail.py`:

```
if rowvar: 
   S = W.transpose()

```
instead of:
```
S = W.transpose()

if not rowvar:
     S = S.transpose()
```
?

**Comment by marcromani** [comment]
Not necessarily. It depends on what "better" means to you. While both code snippets do the same, the latter (original) allows for a better understanding of what is happening in terms of: "at this point in the algorithm it is assumed that rowvar is True (if it's not, matrices X and S are transposed), so at the end of the algorithm, in case that rowvar was in fact not True, transpose S". That's the meaning of the last line. Also, the relation between the H, W matrices of sklearn.decomposition.NMF and our A, S matrices is such that when rowvar is True they are transposed, and I wanted to explicitly write that, too. Finally, transposing is a cheap O(1) operation so speed loss is not a concern.
